### DISCUSSION TITLE
Are custom Providers supported in PowerShell Core and/or macOS & Linux?

### DISCUSSION THREAD

**Discussion by KevinMGranger** [PowerShell/PowerShell#18832] [Q&A] [answered]
There are [docs for writing them](https://learn.microsoft.com/en-us/powershell/scripting/developer/provider/writing-an-item-provider?view=powershell-7.3), but they keep referring to it as Windows PowerShell, and they're under the section `Legacy PowerShell SDK`.

**Comment by jborean93** [comment, marked as answer]
They are still supported with PowerShell although I don't know what package you might need. At a guess I would start with https://www.nuget.org/packages/PowerShellStandard.Library/ and see if it contains the classes you need. There's also [PowerShell.SDK](https://www.nuget.org/packages/Microsoft.PowerShell.SDK/7.4.0-preview.1) but that's more for hosting PowerShell itself and I don't think is necessary.

**Reply by KevinMGranger** [reply]
I just tested on macOS, and it works. It appears that the [Legacy PowerShell SDK docs](https://learn.microsoft.com/en-us/powershell/scripting/developer/provider/windows-powershell-provider-quickstart?view=powershell-7.3) still apply. Thank you!
